source/crud/read-operations/text.txt

```diff
@@ -0,0 +1,260 @@
+<https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions>`_,
+also known as RegEx. While regex is very powerful, and MongoDB even
+offers a `$regex <>` operator, text searches work particularly well when
+applied to unstructured text, like transcripts, essays, or even web
+pages. Because MongoDB's text indexes are aware of concepts like
+plurality, case sensitivity, stop words, whitespace, and punctuation in
+multiple languages, all you have to do is query with a search term to
+find documents that contain your term.
+
+You can only create *one* text index per collection, so every text
+search queries all fields used by the index for term matches.
+
+Examples
+--------
+
+The following examples use the ``movies`` collection in the
+``sample_mflix`` sample database and assume that you have created a text
+index on the ``title`` field.
+
+.. note::
+
+  Text searches require a text index. To create a text index on the
+  ``title`` field of the ``movies`` collection of the ``sample_mflix``
+  database, use the following command on the sample_mflix database:
+
+  .. code-block:: javascript
+
+     db.movies.createIndex({title: "text"})
+
+Query for Words
+~~~~~~~~~~~~~~~
+
+This example queries for Star Trek movies by searching for titles
+containing the word "trek". If you want to query using multiple words,
+separate your words with spaces to query for documents that match any of
+the search terms (logical ``OR``).
+
+.. code-block:: javascript
+
+   const database = client.db("sample_mflix");
+   const collection = database.collection("movies");
+
+   const query = { $text: { $search: "trek" } };
+
+   // Return only the `title` of each matched document
+   const projection = {
+      _id: 0,
+      title: 1
+   };
+
+   // find documents based on our query, sort and projection
+   const cursor = collection
+      .find(query)
+      .project(projection);
+
+   // print a message if no documents were found
+   if ((await cursor.count()) == 0) {
+      console.log("No documents found!");
+   }
+   await cursor.forEach(console.dir);
+
+This operation returns the following documents:
+
+.. code-block:: javascript
+
+   { title: 'Trek Nation' }
+   { title: 'Star Trek' }
+   { title: 'Star Trek Into Darkness' }
+   { title: 'Star Trek: Nemesis' }
+   { title: 'Star Trek: Insurrection' }
+   { title: 'Star Trek: Generations' }
+   { title: 'Star Trek: First Contact' }
+   { title: 'Star Trek: The Motion Picture' }
+   { title: 'Star Trek VI: The Undiscovered Country' }
+   { title: 'Star Trek V: The Final Frontier' }
+   { title: 'Star Trek IV: The Voyage Home' }
+   { title: 'Star Trek III: The Search for Spock' }
+   { title: 'Star Trek II: The Wrath of Khan' }
+
+Success! The query found every document in the ``movies`` collection
+with a title including the word "trek". Unfortunately, it looks like the
+search included one unintended item: "Trek Nation," which is a movie
+about Star Trek and not a Star Trek movie.
```

Comment:
Nice example! Can we make this just a little more self-contained by pointing to the next section?

e.g. "To solve this, we can query with a more specific phrase"

Reply:
Originally this was just one big section that I broke up into different subheadings, will add this to make it flow better.

Reply:
Also added to other sections as well.